modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/UserSetting.kt

```diff
@@ -0,0 +1,162 @@
+package au.com.shiftyjelly.pocketcasts.preferences
+
+import android.content.SharedPreferences
+import kotlinx.coroutines.flow.MutableStateFlow
+
+abstract class UserSetting<T>(
+    protected val sharedPrefKey: String,
+    protected val sharedPrefs: SharedPreferences,
+) {
+
+    var needsSync: Boolean
+        get() = sharedPrefs.getBoolean("${sharedPrefKey}NeedsSync", false)
+        set(value) {
+            sharedPrefs.edit().run {
+                putBoolean("${sharedPrefKey}NeedsSync", value)
+                apply()
+            }
+        }
+
+    fun getSyncValue(): T? =
+        if (needsSync) get() else null
+
+    private val _flow by lazy { MutableStateFlow(get()) }
+    // lazy because the class needs to initialize before calling get()
+    val flow by lazy { _flow }
```

Comment:
Using `by lazy` here feels more like a hacky workaround than a proper solution, but without this the app crashes on startup because it seems to call `get()` when initializing `_flow` before the non-null constructor parameters are present. Let me know if you have any ideas for a better way to handle this.